lib/active_model/serializer/array_serializer.rb

```diff
@@ -5,7 +5,7 @@ class ArraySerializer
       include Enumerable
       delegate :each, to: :@objects
 
-      attr_reader :root, :meta, :meta_key
+      attr_reader :root, :meta, :meta_key, :options, :resource
```

Comment:
I see you're now passing options into the array serializer, but also the resource? Is that so that the adapter can get the resource from it? I think we could get around this if we had pagination as one of the adapter options that gets passed through the serializable resource to the adapter, no?  That way the array serializer doesn't need to 'know' that the json adapter wants to add pagination links.